### PR TITLE
Fix use of asm2wasm imports in side modules

### PR DESCRIPTION
**Pull request by mdboom**
This is something that seems to have broken sometime between 1.37.39 and now.
If a SIDE_MODULE needs one of the helper functions in `asm2wasm`, in this case floating-point remainder, it can't find it at dynamic linking time and fails with:

```
TypeError: import object field 'asm2wasm' is not an Object
```
This solves that by just copying-and-pasting what the MAIN_MODULE gets.  I'm sure there's a better way to avoid this duplication, but I couldn't see it.  Any thoughts?  Happy to fix this to make it right.

### CODE REVIEW

**Comment by mdboom** [comment]
I've added a test here.

File: tests/test_other.py [hunk 1]
```diff
@@ -6197,6 +6203,7 @@ def test_ld_library_path(self):
     self.assertContained('Hello2', out)
     self.assertContained('Hello3', out)
     self.assertContained('Hello4', out)
+    self.assertNotContained('Floating-point remainder failed', out)
```
Comment (kripken) [inline tests/test_other.py]: notContained is somewhat fragile - how about printing "Ok." in an else to the error check, and verifying the that success notification is present?

File: src/support.js [hunk 2]
```diff
@@ -158,7 +158,33 @@ function loadWebAssemblyModule(binary) {
       'Infinity': Infinity,
     },
     'global.Math': Math,
-    env: env
+    env: env,
+    'asm2wasm': { // special asm2wasm imports
```
Comment (kripken) [inline src/support.js]: how about making this and the other place do `'asm2wasm': asm2wasmImports` and we lift the contents out into a singleton in the global scope?

**Comment by mdboom** [comment]
Thanks.  I've updated this based on your comments.

File: src/support.js [hunk 3]
```diff
@@ -85,6 +85,33 @@ function loadDynamicLibrary(lib) {
 }
 
 #if WASM
+var asm2wasmImports = { // special asm2wasm imports
+    "f64-rem": function(x, y) {
```
Comment (kripken) [inline src/support.js]: indentation looks like 4? should be 2 spaces

File: src/support.js [hunk 4]
```diff
@@ -85,6 +85,33 @@ function loadDynamicLibrary(lib) {
 }
 
 #if WASM
+var asm2wasmImports = { // special asm2wasm imports
```
Comment (kripken) [inline src/support.js]: i think this needs to be higher up - it's inside an ifdef for relocatable, but we need this for all wasm code (and causes the CI failures)

**Comment by kripken** [comment]
Tests still look broken here. Let me know if you get stuck on this.

**Comment by kripken** [comment]
Thanks!

Weird there are a bunch of random failures here, but maybe travis had a bad day today... looks good to merge to me.

**Pull request merged**